waltz-test/src/main/python/waltz_ducktape/tests/validation/load_rebalancing_test.py

```diff
@@ -0,0 +1,97 @@
+        Class for tests simulating load rebalancing on server nodes
+    """
+
+    MIN_CLUSTER_SPEC = ClusterSpec.from_list([
+        {'cpu': 1, 'mem': '1GB', 'disk': '25GB', 'additional_disks': {'/dev/sdb': '100GB'}, 'num_nodes': 3},
+        {'cpu': 1, 'mem': '3GB', 'disk': '15GB', 'num_nodes': 2},
+        {'cpu': 1, 'mem': '1GB', 'disk': '25GB', 'num_nodes': 1}])
+
+    def __init__(self, test_context):
+        super(LoadRebalancingTest, self).__init__(test_context=test_context)
+        self.server_cli = ServerCli(self.verifiable_client.nodes[0], self.client_config_path)
+
+    @cluster(cluster_spec=MIN_CLUSTER_SPEC)
+    @parametrize(num_active_partitions=1, txn_per_client=150, num_clients=2, interval=600, timeout=360, delay_before_rebalancing=10)
+    @parametrize(num_active_partitions=4, txn_per_client=100, num_clients=2, interval=600, timeout=360, delay_before_rebalancing=20)
+    def test_rebalancing_on_server_nodes(self, num_active_partitions, txn_per_client, num_clients, interval, timeout, delay_before_rebalancing):
+        validation_cmd = self.client_cli.validate_txn_cmd(self.log_file_path, num_active_partitions, txn_per_client, num_clients, interval)
+        self.run_produce_consume_validate(lambda: self.rebalancing_on_server_nodes(validation_cmd, timeout, num_active_partitions,
+                                                                                   delay_before_rebalancing, txn_per_client * num_clients))
+
+    def rebalancing_on_server_nodes(self, validation_cmd, timeout, num_active_partitions, delay_before_rebalancing, expected_number_of_transactions):
+        """
+        A validate function to simulate load rebalancing on server nodes using preferred-partitions.
+
+        :param validation_cmd: Command that is passed to client node
+        :param timeout: Test timeout
+        :param num_active_partitions: number of active partitions
+        :param delay_before_rebalancing: delay (in seconds) before rebalancing of nodes starts
+        :param expected_number_of_transactions: number of transactions expected to be stored on a waltz storage node
+        """
+
+        storage_admin_port = self.waltz_storage.admin_port
+        server_admin_port = self.waltz_server.port
+        storage_port = self.waltz_storage.port
+        storage_node = self.waltz_storage.nodes[randrange(len(self.waltz_storage.nodes))]
+        random_active_partition = randrange(num_active_partitions)
+
+        # Step 1: Start waltz cluster and execute validation command
+        self.verifiable_client.start(validation_cmd)
+
+        # Step 2: Wait till transactions get processed
+        wait_until(lambda: self.is_max_transaction_id_updated(self.get_host(storage_node.account.ssh_hostname, storage_admin_port),
+                                                              storage_port, random_active_partition, -1), timeout_sec=timeout)
+
+        # Step 3: Wait before staring torture test
+        sleep(delay_before_rebalancing)
```

Comment:
This steps seems unnecessary. As we are anyways waiting for at least one transaction (as per line 53) to be written to the Waltz Cluster.